### DISCUSSION TITLE
Importance of the `width` and `height` for any `next/image` component

### DISCUSSION THREAD

**Discussion by gaborvecsei** [vercel/next.js#45293] [Help] [answered]
*TL;DR*: I would like to know how important it is to use the real image `width` and `height` with the `<Image/>` component.

I store my images locally and then use `getStaticProps` to determine which images should be displayed on the page (calculated at build time).
However, I do not calculate the width and height of the image I just set it to a constant let's say `500`. Is there any problem with this approach? Should I use their real width and height, even when I know that I will visuzlize them smaller with CSS?

Example:

```
        <Image
          src={imageProps.imageUrl}
          alt={imageProps.photoData.photo.description}
          width={500}
          height={500}
          placeholder="empty"
        />

```

**Comment by ciruz** [comment, marked as answer]
It's not important since the image will be distorted.

I answered something similar, maybe this helps you:

With next.js ``13.x`` you can now simply use the ``fill`` attribute, like so:

```javascript
<Image src={image} alt="" fill />
```

result:
![image](https://user-images.githubusercontent.com/987947/209806162-69db51a9-6229-44fb-8631-4301007fcfb0.png)


**If you want to style it, the ``parent`` element must have ``position: relative`` and ``display: block``.**

If you want to control the size of the image, you use the parent element with fill, like so:
```javascript
<div
  style={{
    position: "relative",
    width: "300px",
    height: "100px",
    border: "solid 3px blue",
  }}
>
    <Image src={image} alt="" fill />
</div>
```

result:
![image](https://user-images.githubusercontent.com/987947/209806616-9df49be7-87f0-4de6-b553-81e62ab6cb4d.png)


Or if this is not the desired look, use ``object-fit`` now via ``css/style`` instead of an attribute, like it was the case with next 12.

```javascript
<div
  style={{
    position: "relative",
    width: "300px",
    height: "100px",
    border: "solid 3px blue",
  }}
>
    <Image src={image} alt="" fill style={{ objectFit: "cover" }} />
</div>
```

result:
![image](https://user-images.githubusercontent.com/987947/209806834-aea14a83-07ae-4833-bda6-e6e022fff0db.png)
